NuGetResources/NuGetServiceInfo.cs

```diff
@@ -0,0 +1,103 @@
+﻿//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+namespace Microsoft.PackageManagement.NuGetProvider
+{
+    using Microsoft.PackageManagement.Provider.Utility;
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Parse and store a NuGet endpoint.
+    /// </summary>
+    public class NuGetServiceInfo
+    {
+        /// <summary>
+        /// Maps known @type values to preference attributes. New @type values should be added only after testing.
+        /// </summary>
+        private static readonly Dictionary<string, NuGetServiceInfo> KnownServiceTypeStrings = new Dictionary<string, NuGetServiceInfo>()
+        {
+            { "searchqueryservice", new NuGetServiceInfo(NuGetServiceType.Query, 0) },
+            { "searchqueryservice/3.0.0-beta", new NuGetServiceInfo(NuGetServiceType.Query, 1) },
+            { "searchqueryservice/3.0.0-rc", new NuGetServiceInfo(NuGetServiceType.Query, 2) },
+
+            { "searchautocompleteservice", new NuGetServiceInfo(NuGetServiceType.AutoComplete, 0) },
+            { "searchautocompleteservice/3.0.0-beta", new NuGetServiceInfo(NuGetServiceType.AutoComplete, 1) },
+            { "searchautocompleteservice/3.0.0-rc", new NuGetServiceInfo(NuGetServiceType.AutoComplete, 2) },
+
+            { "registrationsbaseurl", new NuGetServiceInfo(NuGetServiceType.Registrations, 0) },
+            { "registrationsbaseurl/versioned", new NuGetServiceInfo(NuGetServiceType.Registrations, 1) }, // not sure if this will change versions or not, since it uses clientversion property, so it's low in priority
+            { "registrationsbaseurl/3.0.0-beta", new NuGetServiceInfo(NuGetServiceType.Registrations, 2) },
+            { "registrationsbaseurl/3.4.0", new NuGetServiceInfo(NuGetServiceType.Registrations, 3) },
+            { "registrationsbaseurl/3.6.0", new NuGetServiceInfo(NuGetServiceType.Registrations, 4) },
+
+            { "packagebaseaddress/3.0.0", new NuGetServiceInfo(NuGetServiceType.Files, 0) },
+
+            { "reportabuseuritemplate/3.0.0-beta", new NuGetServiceInfo(NuGetServiceType.ReportAbuse, 0) },
```

Comment:
>reportabuse [](start = 15, length = 11)

Curious to know how ReportAbuse functionality is exposed to the users of NuGet provider via PackageManagement cmdlets.

Reply:
This URI is used to build the URL in the returned PackageItem. Access it using $packageItem.Links (where Relationship == 'abuse')

---
In reply to: [154537476](https://github.com/OneGet/NuGetProvider/pull/25#discussion_r154537476) [](ancestors = 154537476)